src/main/java/com/google/roomies/database/FirebaseDatabase.java

```diff
@@ -0,0 +1,158 @@
+    Map<String, Object> data = doc.toMap();
+    this.db.collection(collectionName).add(data);
+  }
+
+  /**
+  * Add a document to a collection as a class.
+  * All document fields must be serializable. The document class must implement
+  * Serializable and have an empty constructor.
+  *
+  * @param collectionName name of collection in Firestore
+  * @param doc document that implements the document interface
+  */
+  @Override
+  public void addDocumentAsClass(String collectionName, Document doc) throws Exception {
+    ApiFuture<DocumentReference> addedDocRef = this.db.collection(collectionName).add(doc);
+    addedDocRef.get().update(TIMESTAMP, FieldValue.serverTimestamp());
+  }
+
+  /**
+  * Update a document with the specified input fields.
+  *
+  * @param collectionName name of collection in Firestore
+  * @param documentID ID of document to update in Firestore
+  * @param fieldsToUpdate a map of <document key to update, new document value>. 
+  */
+  @Override
+  public void updateDocument(String collectionName, String documentID, Map<String, Object> fieldsToUpdate) {
+    DocumentReference docRef = this.db.collection(collectionName).document(documentID);
+    fieldsToUpdate.forEach((fieldName, fieldValue) -> 
+        docRef.update(fieldName, fieldValue));
+  }
+
+  /**
+  * Get a document from Firestore in a map of <key, value>.
+  *
+  * @param collectionName name of collection in Firestore
+  * @param documentID ID of document to get from Firestore
+  */
+  @Override
+  public Map<String, Object> getDocumentAsMap(String collectionName, String documentID) throws Exception {
+    DocumentReference docRef = this.db.collection(collectionName).document(documentID);
+    ApiFuture<DocumentSnapshot> future = docRef.get();
+    DocumentSnapshot document = future.get();
+    if(document.exists()) {
+      return document.getData();
+    } else {
+      throw new Exception("Error getting document");
+    }
+  }
+
+  /**
+  * Get all documents with the input field value.
+  *
+  * @param collectionName name of collection in Firestore
+  * @param field document field to search
+  * @param fieldValue value of field
+  */
+  @Override
+  public List<QueryDocumentSnapshot> getDocumentsWithFieldValue(
+      String collectionName, String field, Object fieldValue) throws Exception {
+    ApiFuture<QuerySnapshot> future =
+      db.collection(collectionName).whereEqualTo(field, fieldValue).get();
+    List<QueryDocumentSnapshot> documents = future.get().getDocuments();
+    return documents;
+  }
+  
+  /**
+  * Get all documents in specified collection.
+  *
+  * @param collectionName name of collection in Firestore
+  */
+  @Override
+  public List<QueryDocumentSnapshot> getAllDocumentsInCollection(String collectionName) throws Exception { 
+    ApiFuture<QuerySnapshot> future = db.collection(collectionName).get();
+    List<QueryDocumentSnapshot> documents = future.get().getDocuments();
```

Comment:
Calling get on a Future is /usually/ a bad idea in production code - the point of a future is that you don't have a blocking operation until you need it. Did you consider using Futures in the API itself?